### DISCUSSION TITLE
accessing slider values as it moves

### DISCUSSION THREAD

**Discussion by gorkemer** [jspsych/jsPsych#1452] [Q&A] [answered]
Hello everyone,

I am trying to access the value of the slider position as the observers move the slider. I can get that value number once they submitted a response with the 

```
display_element.querySelector('#jspsych-canvas-slider-response-next').addEventListener('click', function () {

        response = display_element.querySelector('#jspsych-canvas-slider-response-response').valueAsNumber;
                
        });
```
, however, is there a way to get the immediate slider value without requiring a response submission? 

Thanks!
Gorkem

**Comment by jodeleeuw** [comment, marked as answer]
Hi Gorkem,

You can add an event listener for the `input` event instead. See [this helpful article](https://www.impressivewebs.com/onchange-vs-oninput-for-range-sliders/).

**Reply by gorkemer** [reply]
That works. Thank you, Josh. 

Here is the code for anyone interested:

```
        display_element.querySelector('#jspsych-canvas-slider-response-response').addEventListener('input', function () {
        var slider_value = document.querySelector('input').value;
        console.log(slider_value) // to see it on the console
        });
```
